python/packages/autogen-core/tests/test_utils/__init__.py

```diff
@@ -46,3 +55,31 @@ def __init__(self) -> None:
 
     async def on_message(self, message: Any, ctx: MessageContext) -> Any:
         raise NotImplementedError
+
+
+@default_subscription
+class MyAgent(RoutedAgent):
+    def __init__(self, name: str) -> None:
+        super().__init__("My agent")
+        self._name = name
+        self._counter = 0
+
+    @message_handler
+    async def my_message_handler(self, message: MyMessage, ctx: MessageContext) -> None:
+        self._counter += 1
+        if self._counter > 5:
+            return
+        content = f"{self._name}: Hello x {self._counter}"
+        await self.publish_message(MyMessage(content=content), DefaultTopicId())
+
+
+@default_subscription
+class CounterAgent(RoutedAgent):
```

Comment:
This is very similar to the loopback agent above. Since we are only using it for one specific test, perhaps it makes sense to move it next to the test instead?